evidence/pool.go

```diff
@@ -171,6 +223,53 @@ func (evpool *Pool) removeEvidence(
 	}
 }
 
+func (evpool *Pool) cleanupValToLastHeight(blockHeight int64) {
```

Comment:
As we bound the expiration of evidence by both block height and time, this function should also remove the validator if the latest height is of a height which has expired (in relation to time).

Reply:
If we implement this "is the validator still slashable" map properly then this can also become a proxy for checking for expiration and rather then have to check the params if the evidence is expired we just check whether the validators of the evidence are still punishable

Reply:
Note that this highlights an incongruity in our thinking in the sense that it's possible that a validator who might be part of an evidence that is now expired could still be part of the validator set and therefore still be slashed.